### PR TITLE
XIVY-15542 executes mining action after jump into sub

### PR DESCRIPTION
**Pull request by ivy-fhe**
- Fetches data once and stores it in injectable
- Dispatches MiningAction after JumpAction

### CODE REVIEW

File: src/process-mining-visualisation/di.config.ts [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-import { SetMiningActionHandler } from './action';
+import { MiningActionHandler, SetMiningActionHandler } from './action';
 import { DiagramCaption } from './DiagramCaption';
 import { DiagramCaptionView } from './DiagramCaptionView';
 import { MiningLabelView } from './MiningLabelView';
 import { MiningLabel } from './MiningLabel';
 
 export const ivyMiningModule = new FeatureModule((bind, _unbind, isBound) => {
+  configureActionHandler({ bind, isBound }, UpdateModelAction.KIND, MiningActionHandler);
+  configureActionHandler({ bind, isBound }, SetModelAction.KIND, MiningActionHandler);
```
Comment (ivy-lli) [inline src/process-mining-visualisation/di.config.ts:18]: why can't you handle this in the setMiningActionHandler?

File: src/process-mining-visualisation/mining-action.ts [hunk 2]
```diff
@@ -57,7 +44,15 @@ export class MiningCommand extends Command {
...
     const model = context.root;
-    this.action.data.nodes.forEach(node => {
+    if (model.children.filter(e => e.type === DiagramCaption.TYPE).length > 0) {
+      return model;
+    }
+    return this.populate(model);
+  }
+
+  async populate(model: SModelRootImpl) {
+    const data = await (await fetch(this.miningData.url)).json();
```
Comment (ivy-lli) [inline src/process-mining-visualisation/mining-action.ts:54]: json() returns an any type if I'm not wrong. 
So I would recommend that you don't delete the types above but cast it here so you're editor can help you to not write typos...

File: src/process-mining-visualisation/action.ts [hunk 3]
```diff
@@ -1,17 +1,9 @@
...
-
-  handle(action: Action): Action | void {
-    if (MiningAction.is(action.kind)) {
-      this.modelInitializationConstraint.onInitialized(() => {
-        this.feedbackDispatcher.registerFeedback(this, [action]);
-      });
-    }
-  }
+  @inject(GLSPActionDispatcher) protected actionDispatcher: GLSPActionDispatcher;
+  handle = (_: Action): void => this.actionDispatcher.dispatchAfterNextUpdate(MiningAction.create());
```
[the remark this answers is not in the document]
Comment (ivy-fhe) [reply src/process-mining-visualisation/action.ts:8]: I wrote it like this so it is clear that handle receives an action but does not use it

The "dispatchAfterNextUpdate" is no longer needed :smile:

File: src/process-mining-visualisation/mining-action.ts [hunk 4]
```diff
@@ -7,7 +7,11 @@ import { MiningLabel } from './MiningLabel';
 
 export interface MiningAction extends Action {
   kind: typeof MiningAction.KIND;
-  data: MiningData;
+}
+
+export interface period {
```
[the remark this answers is not in the document]
Comment (ivy-fhe) [reply src/process-mining-visualisation/mining-action.ts:12]: Fixed in  [Cleanup](https://github.com/axonivy/process-miner-viewer/pull/14) :wink:

**Pull request merged**